.NET/Microsoft.Recognizers.Text/Culture.cs

```diff
@@ -29,15 +32,34 @@ public sealed class Culture
             new Culture("Korean", Korean)
         };
 
+        private static readonly string[] SupportedCultureCodes = SupportedCultures.Select(c => c.CultureCode).ToArray();
+        
         private Culture(string cultureName, string cultureCode)
         {
             this.CultureName = cultureName;
             this.CultureCode = cultureCode;
         }
 
-        public static string[] GetSupportedCultureCodes()
+        // If the input culture code isn't in the supported cultures list,
+        // the first culture with the same prefix as the input one will be returned.
+        // Otherwise, the original input culture code will be returned.
+        // e.g. "en-029"->"en-*" "vo-id"->"vo-id" "en-us"->"en-us"
+        public static string GetCultureCodeWithFallback(string cultureCode)
```

Comment:
I prefer not to change the method name. It's goal is still the same. 
We may even specify that fr-, es-, pt- are also *.